tools: add additional version checks to changelog validation

dhwaniserai opened the pull request:
Add a check to verify that last two versions are not same

<!--
Tips:
- Please read CONTRIBUTING.md to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->


**Description of changes:**

Added the following checks to `CHANGELOG` validation
- All release versions are unique
- All release versions in `CHANGELOG` are in descending order

**Testing done:**

- Validation passed with current `CHANGELOG.md`
- Validation failed with duplicate kit versions
- Validation failed with version in non descending order eg: v4.0.2 before v4.0.1

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

ginglis13 reviewed (approved):
Repeated your testing steps, LGTM

dhwaniserai committed:
tools: add additional version checks to changelog validation

Added checks for version format consistency and proper semantic versioning patterns
to ensure changelog entries follow the expected format standards.

Signed-off-by: Dhwani Serai <dhwanise@amazon.com>

dhwaniserai commented:
force push refers to the following change in validation checks
- Make sure that versions are in descending order
- All versions are unique (no duplicates)

dhwaniserai commented:
force push refers to a small change from shellcheck

arnaldo2792 commented on tools/validate-changelog.sh:
```diff
@@ -4,9 +4,22 @@
+if ! echo "${versions}" | sort -V -r | diff - <(echo "${versions}"); then
+  echo "CHANGELOG validation FAILED! Versions must be in descending order."
+  exit 1
+else
+  echo "CHANGELOG validation PASSED!"
```
You can avoid the last branch, and just echo "SUCCESS" at the end of this script.

arnaldo2792 commented on tools/validate-changelog.sh:
```diff
@@ -4,9 +4,22 @@
+fi
+
+# check that versions are in descending order
+if ! echo "${versions}" | sort -V -r | diff - <(echo "${versions}"); then
```
Can you add a comment explaining what each part of the command does? I don't understand what `diff - ` at the end does.

dhwaniserai commented:
force push refers to changes resolving the feedback above^

arnaldo2792 reviewed (approved):
:boat:

ginglis13 reviewed (approved):
Checked out the PR locally, LGTM
